### ISSUE TITLE
machine/mount: prefetch failure renders mount unusable

### ISSUE THREAD

**Issue by rjeczalik** [koding/koding#11124]
```
$ kd mount green_coconut:~/koding
Mounting to /Users/rjeczalik/koding/mnt/development directory.
Checking remote path...
Mounted remote directory /home/rjeczalik/koding has 124853 file(s) of total size 2.4 GiB
Initializing mount /home/rjeczalik/koding -> /Users/rjeczalik/koding/mnt/development...

Using git files strategy to prefetch initial mount data.
Copying files: 100.0% (1534/1534), 324 MiB/324 MiB | 1.7 MiB/s
Cannot prefetch mount files: git command exited with error: exit status 128 ("error: packfile .git/objects/pack/pack-3291f55478f854d5e5117a4758a35c83978c26c1.pack does not match index\nerror: packfile .git/objects/pack/pack-3291f55478f854d5e5117a4758a35c83978c26c1.pack do
es not match index\nfatal: unable to read tree e90ddb32d5464c024514b08725f62407f7f0698d\n")
Created mount with ID: c455d936-7470-4f21-92a2-8965b132e293
```

```
$ cd ~/koding/mnt/development/
error: packfile .git/objects/pack/pack-3291f55478f854d5e5117a4758a35c83978c26c1.pack does not match index
error: packfile .git/objects/pack/pack-3291f55478f854d5e5117a4758a35c83978c26c1.pack does not match index
fatal: unable to read tree e90ddb32d5464c024514b08725f62407f7f0698d
$
```

To solve that make prefetch failure critical causing mount to fail. Cleanup the cache and mount directory afterwards. After the mount is proven to be able to recover from prefetch problems, make the prefetch failure a soft error again.

### Signature
```
2f726a65637a616c696b2f31336430303064382d663136612d343330312d613936332d3334626134636364336636342f6275672f346339376433623833383135646335303338336538323338353434656366326235346230663462342e6a736f6e2e677a2e30
```

**Comment by rjeczalik** [comment]
I think this could have happened because the files / repo was in use on remote (koding service was started while the rsync copied the files over). Did not confirm yet, just a hint.

**Comment by ppknap** [comment]
Interesting:

```
Copying files: 100.0% (1534/1534), 324 MiB/324 MiB | 1.7 MiB/s
```

as far as I remember, koding bare repository has around ~800-900MB. Maybe we start mounting(rsync scanned list of files) when repository was not downloaded yet. 

This may be solved by recursively checking for `.git/objects/pack/tmp_pack_*` files inside created remote index(first mount step) and returning error when such files exist. wdyt?

**Comment by rjeczalik** [comment]
> wdyt?

After I dug more into this, I discovered that my .git on remote was corrupted (thus 300MB instead of 900MB). Not sure how that happened (rsync started to sync on a half-prefetched cache?), but I've cloned my repo again.

For now I think the best solution would be to not allow for using the mount if prefetched failed for whatever reason. A safe measure.

Then we could be adding filesystem checks and tests and eventually enable the mounts on failed prefetches again, as in description.

**Comment by ppknap** [comment]
```
Not sure how that happened (rsync started to sync on a half-prefetched cache?)
```
It's possible, since rsync could prevent git from removing files that had been downloaded.

I agree that we should fail mounts when prefetch fails(at least for now). However, if it's rsync that corrupts remote clone, we should also implement checkers like the above, because even if we fail prefetch, the remote repository will be already corrupted.

**Comment by ppknap** [comment]
These checks could also solve #11125

**Comment by ppknap** [comment]
Fixed by #11134 & #11156
